php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Default class for handling file uploads in LORIS
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Provides a default handler for files that were uploaded on a LORIS
+ * web site using LorisForm.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class DefaultFilesUploadHandler implements FilesUploadHandler
+{
+    /**
+     * Default directory to move the uploaded files to if none is explicitely
+     * specified.
+     */
+    public static $DEFAULT_MOVE_DIR = __DIR__ . "../user_uploads";
+    /**
+     *  The LorisForm used to upload the file(s).
+     */
+    private $_lorisForm;
+    /**
+     * Directory to move the uploaded files to. Will be created if it does not
+     * exist.
+     */
+    private $_moveDirPath;
+    /**
+     * When creating $_moveDirPath, these are the access rights to give it
+     * (should be an octal number).
+     */
```

Comment:
Is it a string representing an octal number or is it an int? If the latter, the parenthesis is meaningless. Numbers don't have number bases as they're passed around in memory.

Reply:
The comment was more an indication of the literal representation that should be used for this number so that the calling code is easy to read and very UNIX-like. 0755 and 493 (0755 in decimal form) are valid arguments for this method and produce the same result, but 0755 is easier to read. I can remove the comment if you think it is misleading though.